include/dxc/DxilRootSignature/DxilRootSignature.h

```diff
@@ -388,6 +385,30 @@ bool VerifyRootSignature(_In_ const DxilVersionedRootSignatureDesc *pDesc,
                          _In_ llvm::raw_ostream &DiagStream,
                          _In_ bool bAllowReservedRegisterSpace);
 
+class ScopedVersionedRootSignature {
+  const DxilVersionedRootSignatureDesc *m_pRootSignature;
+
+public:
+  ScopedVersionedRootSignature() : m_pRootSignature(nullptr) {}
+  ScopedVersionedRootSignature(
+      const DxilVersionedRootSignatureDesc *pRootSignature)
+  : m_pRootSignature(pRootSignature){}
+  ~ScopedVersionedRootSignature() { 
+      DeleteRootSignature(m_pRootSignature);
+  }
+  const DxilVersionedRootSignatureDesc* operator -> () const {
+    return m_pRootSignature;
+  }
+  const DxilVersionedRootSignatureDesc ** get_address_of() {
```

Comment:
Can this use the `&` operator?  Or is that now considered bad practice?

Also, could the functions take the new `ScopedVersionedRootSignature` instead of `const DxilVersionedRootSignatureDesc**`?

And, could `ScopedVersionedRootSignature` just be `DxilVersionedRootSignature`?

Reply:
I was following the example of ComPtr, which intentionally doesn't allow & access. I presume that's to make it a bit more explicit and so harder to accidentally take the address of and then overwrite a non-zero pointer.

Reply:
Yes, those functions could be converted but that's a bigger-scope change than I was hoping for here.

Reply:
Re: `&` operator:  Strangely, `CComPtr` supports it, we use that all over the place, and it does throw (or assert?) if the pointer is not null.

On that note, we should probably fail somehow if the pointer is not null here.  Perhaps assert and return nullptr (for NDEBUG build)?  That way we don't silently leak if it's misused.

Reply:
Should that be a DXASSERT or is a different macro more appropriate in this header?

Reply:
(I did think of adding one but wasn't sure what macro to use :-)) )

Reply:
Added not-null check